torch_frame/config/text_model.py

```diff
@@ -0,0 +1,22 @@
+from dataclasses import dataclass
+from typing import Callable
+
+from torch import Tensor
+
+from torch_frame.typing import TensorData
+
+
+@dataclass
+class TextModelConfig:
+    r"""Text model that maps a dictionary of :class:`MultiNestedTensor`
+    into embeddings of each row.
+
+    Args:
+        model (callable): A callable text model that takes a dictionary
+            of :class:`MultiNestedTensor` as input and outputs embeddings
+            for each row in the :class:`MultiNestedTensor`.
+        out_channels (int): Text model output channels.
+
+    """
+    model: Callable[[TensorData], Tensor]
+    out_channels: int
```

Comment:
Can we be more explicit about type annotation? What's the input shape of multi-nested tensor (should be `[batch_size, 1, *]`) and output shape of Tensor (should be `[batch_size, 1, out_channels]`).

Reply:
`model: Callable[[dict[MultiNestedTensor]], Tensor]`